### PR TITLE
Refactor the Tooltip component and use modern @floating-ui/react library instead of react-popper

### PR DESCRIPTION
**Pull request by axeleriksson147**
This PR refactors the Tooltip component and uses the sequel to `react-popper`, `@floating-ui/react` - https://github.com/floating-ui/floating-ui.

- The code is easier to understand.
- Less libraries involved.
- Easier to do component testing.

All tests pass and I've tested and compared the components manually against the old Storybook and the component behaves the same or better.

### CODE REVIEW

File: packages/components/package.json [hunk 1]
```diff
@@ -36,19 +36,18 @@
     "node": ">=14.21.2"
   },
   "dependencies": {
+    "@emotion/cache": "^11.11.0",
+    "@emotion/react": "^11.11.4",
```
Comment (axeleriksson147) [inline packages/components/package.json:40]: These dependencies were used in the `StylesInjection.tsx` but not explicitly added as dependencies.

File: packages/styles/src/components/atoms/_hint-or-tooltip.scss [hunk 2]
```diff
@@ -1,41 +0,0 @@
-.tk-hint-or-tooltip {
```
Comment (axeleriksson147) [inline packages/styles/src/components/atoms/_hint-or-tooltip.scss:1]: Provided by `@floating-ui/react`.

File: packages/components/src/components/tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -65,140 +53,123 @@ const Tooltip: React.FC<TooltipProps> = ({
...
-  return (
-    <>
-      { /**
-       * The element tooltip is wrapped around.
-       */ }
-      { /** Extra span to make sure Tooltip works on disabled children */ }
-      {displayTrigger === 'hover' &&
-          showTooltipOnHover(children, handleMouseMove)}
-      {displayTrigger === 'click' &&
-          showTooltipOnClick(children, showClick, setShowClick)}
```
Comment (axeleriksson147) [inline packages/components/src/components/tooltip/Tooltip.tsx:126]: Before clicking anywhere on a hint would close it if displayTrigger was `click` which is incorrect behaviour. Now it'll only close if close label is clicked, the reference element or if you click outside.

File: packages/components/src/components/tooltip/Tooltip.tsx [hunk 4]
```diff
@@ -65,140 +53,123 @@ const Tooltip: React.FC<TooltipProps> = ({
...
   type: 'hint',
 };
 
 Tooltip.propTypes = {
+  className: PropTypes.string,
   children: PropTypes.any,
   closeLabel: PropTypes.string,
   description: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
   displayTrigger: PropTypes.oneOf(['click', 'hover']),
+  hoverDelay: PropTypes.number,
```
Comment (antoinerollindev) [inline packages/components/src/components/tooltip/Tooltip.tsx:168]: Don't we miss hoverTimeout here? 🤔

**Review by antoinerollindev — APPROVED** [review]

**Pull request merged**